### PR TITLE
Add frontend testing, require node 12

### PR DESCRIPTION
**Pull request by silverwind**
- Add basic frontend unit testing infrastructure using jest in [ESM mode](https://jestjs.io/docs/ecmascript-modules)
- Rename 'make test' to 'make test-backend'
- Introduce 'make test-frontend' and 'make test' that runs both
- Bump Node.js requirement to v12. v10 will be EOL in less than a month.
- Convert all build-related JS files to ESM.

I opted to run frontend tests run as part of the compliance pipeline because they complete fast and are not platform-specific like the golang tests.

`make test` seems to never be triggered in drone, so this rename only affects users.

## :warning: BREAKING :warning: 

The minimum required node version for compiling the frontend has increased to v12.

### CODE REVIEW

**Comment by silverwind** [comment]
Labeling as breaking because of the bump to the minimum Node.js version.

File: docs/config.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ params:
   version: 1.13.6
   minGoVersion: 1.14
   goVersion: 1.16
-  minNodeVersion: 10.13
+  minNodeVersion: 12.17.0
```
Comment (silverwind) [inline docs/config.yaml]:
```suggestion
  minNodeVersion: 12.17
```

**Comment by zeripath** [comment]
do we actually need to up the required version of node for this change?

**Comment by 6543** [comment]
bump of node version in docs is needed too

**Comment by silverwind** [comment]
> do we actually need to up the required version of node for this change?

Yes, bump is kind of required unless I want to opt to configure jest in CJS (legacy) mode as the ESM mode requires [a feature](https://nodejs.org/api/vm.html#vm_class_vm_module) only present in Node 12.16.0 or above.

> bump of node version in docs is needed too

I think I handled this with the change in `docs/config.yaml`, or did I miss anything?

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
node eof ref: https://nodejs.org/en/about/releases/

**Review by zeripath — APPROVED** [review]
Will probably need to wire this into drone

**Comment by silverwind** [comment]
Drone is already done: https://drone.gitea.io/go-gitea/gitea/38171/1/9

It's executed as part of the compliance pipeline because it's not platform dependant like the go tests are. Maybe we should move it to the actual test pipelines later for consistency.

**Comment by zeripath** [comment]
So unfortunately this has broken dropzone - possibly others.

See https://try.gitea.io/fnetx/everything/issues/1

```
index.js?v=d613bc151b1bb2009cfe73a98898d3c0:206 jQuery.Deferred exception: i is not a constructor TypeError: i is not a constructor
    at https://try.gitea.io/js/index.js?v=d613bc151b1bb2009cfe73a98898d3c0:148:1585
    at Generator.next (<anonymous>)
    at f (https://try.gitea.io/js/index.js?v=d613bc151b1bb2009cfe73a98898d3c0:148:1153) undefined
a.Deferred.exceptionHook @ index.js?v=d613bc151b1bb2009cfe73a98898d3c0:206
dropzone.js:6 Uncaught Error: No URL provided.
    at new c (dropzone.js:6)
    at dropzone.js:6
    at Function.oe.discover (dropzone.js:6)
    at oe._autoDiscoverFunction (dropzone.js:8)
    at HTMLDocument.z (dropzone.js:8)
index.js?v=d613bc151b1bb2009cfe73a98898d3c0:206 Uncaught TypeError: i is not a constructor
    at index.js?v=d613bc151b1bb2009cfe73a98898d3c0:148
    at Generator.next (<anonymous>)
    at f (index.js?v=d613bc151b1bb2009cfe73a98898d3c0:148)
```

Fixing the dropzone.js to read: 
```js
export default async function createDropzone(el, opts) {
  const [{default: Dropzone}] = await Promise.all([
    import(/* webpackChunkName: "dropzone" */'dropzone'),
    import(/* webpackChunkName: "dropzone" */'dropzone/dist/dropzone.css'),
  ]);

  Dropzone.Dropzone.autoDiscover = false;
  return new Dropzone.Dropzone(el, opts);
}
``` 
makes this one stop - but I suspect there are other potential issues.

**Pull request merged**